### ISSUE TITLE
Possible manipulation by malicious users

### ISSUE THREAD

**Issue by shuo-young** [saucepoint/v4-stoploss#8]
Dear saucepoint:
  I find that a potential risk can arise from the possible parameter manipulation of the function _afterSwap()_. Malicious attackers can input any value of parameters _key_ and _params_  to manipulate intermediate variables and the key stop loss operation in the internal function _fillStopLoss()_ can also be influenced. 

  I think critical hooks should all be protected which can only be invoked by pool manager. For example, the afterInitialize() is protected in your StopLoss contract. However, the _afterSwap()_ isn't. So adding a simple protective modifier may be good  to avoid potential risk.

  Hope it helps and glad to discuss further.

**Comment by saucepoint** [comment]
yep has been flagged before -- just havent had the time to update it. thanks for keeping an eye out!

will leave this open until its fixed 😄
